File: C#/SchoolSystem/SchoolSystem.Core/Entitites/CourseGradebook.cs
// Code scaffolded by EF Core assumes nullable reference types (NRTs) are not used or disabled.
// If you have enabled NRTs for your project, then un-comment the following line:
// #nullable disable

namespace SchoolSystem.Core.Entities
{
    public partial class CourseGradebook
    {
        public int CourseGradebookId { get; set; }
        public int? StudentId { get; set; }
        public decimal Q1 { get; set; }
        public decimal Q2 { get; set; }
        public decimal Q3 { get; set; }
        public decimal? Average { get; set; }
        public int? TeacherId { get; set; }
        public int? CourseId { get; set; }
        public bool Status { get; set; } = false;

        public virtual Course Course { get; set; }
        public virtual Student Student { get; set; }
        public virtual Teacher Teacher { get; set; }
    }
}

File: Java/schoolsystem/src/main/java/org/com/schoolsystem/core/entities/CourseGradeBook.java
package org.com.schoolsystem.core.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;

@Entity
@Table(name = "tb_course_gradebook")
@Data
public class CourseGradeBook {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "coursegradebookid")
    public int courseGradebookId;
    @Column(name = "studentid")
    private int studentId;
    private double q1;
    private double q2;
    private double q3;
    private double average;
    @Column(name = "teacherid")
    private int teacherId;
    @Column(name = "courseid")
    private int courseId;
    private boolean status;

}
